### PR TITLE
support k8s websockets

### PR DESCRIPTION
**Pull request by christianvogt**
Closes: https://github.com/opendatahub-io/odh-dashboard/issues/1138

## Description

Adds support for `dynamic-plugin-sdk` `useK8sWatchResource` using web sockets.

Note that this PR only adds support for web sockets but does not introduce any usage of web sockets to the application.

Backend:
- Bump `fastify` to latest versions.
- Add dependency `@fastify/websocket`.
- Introduce a websocket proxy for route `/wss/k8s`.
- Removed the work around for `safeURLPassThrough` as the related bug has been fixed in the `dynamic-plugin-sdk`.
- Created a work around for https://issues.redhat.com/browse/RHCLOUD-28058
  - Return an empty response after successfully deleting a k8s resource.

Frontend:
- Bumps `dynamic-plugin-sdk` and `@openshift/dynamic-plugin-sdk-utils` to latest versions.
 - This gives us access to the latest APIs which allow for static models to be used to bypass API discovery. 
- Isolates the dynamic-plugin-sdk redux store from our own by creating a custom redux context for ODH dashboards.
- Add dev webpack proxy for websockets.

## How Has This Been Tested?

Ad hoc testing with custom test page.
Smoke test current app.

## Test Impact

None.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by lucferbux** [comment]
> /hold Target branch is TBD.

I would say that this should go to something like `f/performance` since that's the main goal of it, let me create a tracker to sum up all the performance work we have at the moment.
@andrewballantyne What do you think?

**Comment by andrewballantyne** [comment]
That works for me -- we should also talk about our merging strategy more -- I demoed a loose idea this morning but the process needs work. Things that are not feature roadmap items can't live in `incubation` forever. But that feature branch can be done now and we can work to improve the process behind the scenes.

File: backend/src/app.ts [hunk 1]
```diff
@@ -4,8 +4,10 @@ import { LOG_DIR } from './utils/constants';
 import fastifyStatic from '@fastify/static';
 import fastifyAutoload from '@fastify/autoload';
 import fastifySensible from '@fastify/sensible';
+import fastifyWebsocket from '@fastify/websocket';
 import { FastifyInstance } from 'fastify/types/instance';
 
+console.log('starting');
```
Comment (lucferbux) [inline backend/src/app.ts:10]: Do we need a log here?

**Comment by lucferbux** [comment]
> That works for me -- we should also talk about our merging strategy more -- I demoed a loose idea this morning but the process needs work. Things that are not feature roadmap items can't live in `incubation` forever. But that feature branch can be done now and we can work to improve the process behind the scenes.

Yes, we can have a sprint for stability or something and then merge the feature branch, I'm open to any idea here.

**Comment by lucferbux** [comment]
First rundown seems ok, I'm gonna try to get some time today or tomorrow to give a deeper look just in case.

File: backend/src/app.ts (hunk 1 above)
Comment (christianvogt) [reply backend/src/app.ts]: removed

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm
Tested, running successfully, nothing is broken.

**Pull request merged**